apps/framework-cli/src/framework/core/infrastructure_map.rs

```diff
@@ -1392,22 +1429,138 @@ impl PartialInfrastructureMap {
                     columns: partial_table.columns.clone(),
                     order_by: partial_table.order_by.clone(),
                     deduplicate: partial_table.deduplicate,
-                    version: partial_table
-                        .version
-                        .clone()
-                        .map(Version::from_string)
-                        .unwrap_or_else(|| Version::from_string("1.0.0".to_string())),
-                    source_primitive: partial_table.source_primitive.clone().unwrap_or_else(|| {
-                        PrimitiveSignature {
-                            name: partial_table.name.clone(),
-                            primitive_type: PrimitiveTypes::DataModel,
-                        }
-                    }),
+                    version: Version::from_string("0.0".to_string()),
+                    source_primitive: PrimitiveSignature {
+                        name: partial_table.name.clone(),
+                        primitive_type: PrimitiveTypes::DataModel,
+                    },
                 };
                 (id.clone(), table)
             })
             .collect()
     }
+
+    fn convert_topics(&self) -> HashMap<String, Topic> {
+        self.topics
+            .iter()
+            .map(|(id, partial_topic)| {
+                let topic = Topic {
+                    name: partial_topic.name.clone(),
+                    columns: partial_topic.columns.clone(),
+                    retention_period: std::time::Duration::from_secs(
+                        partial_topic.retention_period,
+                    ),
+                    partition_count: partial_topic.partition_count,
+                    version: Version::from_string("0.0".to_string()),
+                    source_primitive: PrimitiveSignature {
+                        name: partial_topic.name.clone(),
+                        primitive_type: PrimitiveTypes::DataModel,
+                    },
+                };
+                (id.clone(), topic)
+            })
+            .collect()
+    }
+
+    fn convert_api_endpoints(&self, language: SupportedLanguages) -> HashMap<String, ApiEndpoint> {
```

Comment:
Are we changing the api for this as well?

Reply:
Sorry, I don't quite understand. Could you elaborate?

Reply:
The current way of declaring consumption apis is through on disk layout. That's read through the rust part. Are we changing the API to declare consumption apis?

that might be the right thing to do, I am not sure I saw the paper prototypes there, hence the question

Reply:
I believe the plan is to keep egress the same because the ux doesn't change much. We'd need to reconcile ingress & egress since they're read in different ways though.